### PR TITLE
azuredatastudio: 1.17.1 -> 1.20.0

### PR DESCRIPTION
**Pull request by xavierzwirtz**
###### Motivation for this change
Azure Data Studio needed updated, and a .desktop file created.

###### Things done
Updated azure data studio to 1.20.0 and added a .desktop file.

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by xavierzwirtz** [comment]
cc @husseinraoouf

File: pkgs/applications/misc/azuredatastudio/azuredatastudio.desktop [hunk 1]
```diff
@@ -0,0 +1,18 @@
+[Desktop Entry]
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/azuredatastudio/azuredatastudio.desktop:1]: Please use the appropriate function to create this file and place it inline with the default.nix.

File: pkgs/applications/misc/azuredatastudio/default.nix [hunk 2]
```diff
@@ -21,11 +21,11 @@
 stdenv.mkDerivation rec {
 
   pname = "azuredatastudio";
-  version = "1.17.1";
+  version = "1.20.0";
 
   src = fetchurl {
-    url = "https://azuredatastudiobuilds.blob.core.windows.net/releases/${version}/azuredatastudio-linux-${version}.tar.gz";
-    sha256 = "0px9n9vyjvyddca4x7d0zindd0dim7350vkjg5dd0506fm8dc38k";
+    url = "https://sqlopsbuilds.azureedge.net/stable/cfbadd1eaef2dfa1fd643009c572137c704f1009/azuredatastudio-linux-${version}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/azuredatastudio/default.nix:27]: Is there a stable url which we could use?

File: pkgs/applications/misc/azuredatastudio/default.nix [hunk 3]
```diff
@@ -60,7 +60,7 @@ stdenv.mkDerivation rec {
   ];
 
   # this will most likely need to be updated when azuredatastudio's version changes
-  sqltoolsservicePath = "${targetPath}/resources/app/extensions/mssql/sqltoolsservice/Linux/2.0.0-release.56";
+  sqltoolsservicePath = "${targetPath}/resources/app/extensions/mssql/sqltoolsservice/Linux/3.0.0-release.4";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/azuredatastudio/default.nix:63]: Add a comment next to version to check this with each update.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 93292` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>azuredatastudio</li>
  </ul>
</details>

**Pull request closed without merging**